src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -0,0 +1,706 @@
+        /// <returns><see langword="true" /> if the parse was successful, <see langword="false" /> otherwise.</returns>
+        public static bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider, out Half result)
+        {
+            bool ret = false;
+            if (float.TryParse(s, style, provider, out float floatResult))
+            {
+                result = (Half)floatResult;
+                ret = true;
+            }
+            else
+            {
+                result = default(Half);
+            }
+            return ret;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool AreZero(Half left, Half right)
+        {
+            // IEEE defines that positive and negative zero are equal, this gives us a quick equality check
+            // for two values by or'ing the private bits together and stripping the sign. They are both zero,
+            // and therefore equivalent, if the resulting value is still zero.
+            return (ushort)((left.m_value | right.m_value) & ~SignMask) == 0;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsNaNOrZero(Half value)
+        {
+            return ((value.m_value - 1) & ~SignMask) >= PositiveInfinityBits;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static ushort StripSign(Half value)
+        {
+            return (ushort)(value.m_value & ~SignMask);
+        }
+
+        /// <summary>
+        /// Compares this object to another object, returning an integer that indicates the relationship.
+        /// </summary>
+        /// <returns>A value less than zero if this is less than <paramref name="obj"/>, zero if this is equal to <paramref name="obj"/>, or a value greater than zero if this is greater than <paramref name="obj"/>.</returns>
+        /// <exception cref="ArgumentException">Thrown when <paramref name="obj"/> is not of type <see cref="Half"/>.</exception>
+        public int CompareTo(object? obj)
+        {
+            if (!(obj is Half))
+            {
+                return (obj is null) ? 1 : throw new ArgumentException(SR.Arg_MustBeHalf);
+            }
+            return CompareTo((Half)(obj));
+        }
+
+        /// <summary>
+        /// Compares this object to another object, returning an integer that indicates the relationship.
+        /// </summary>
+        /// <returns>A value less than zero if this is less than <paramref name="other"/>, zero if this is equal to <paramref name="other"/>, or a value greater than zero if this is greater than <paramref name="other"/>.</returns>
+        public int CompareTo(Half other)
+        {
+            if ((short)(m_value) < (short)(other.m_value))
```

Comment:
This isn't going to be right as it is ignoring `NaN` and `0` vs `-0`

Reply:
Oh, I thought I had fixed this. Good catch. Fixed now.